src/pages/user/institution/InstitutionCard.tsx

```diff
@@ -43,45 +37,15 @@ interface InstitutionCardProps {
 
 const InstitutionCard: FC<InstitutionCardProps> = ({ orgunitId, setAffiliationIdToRemove }) => {
   const { t } = useTranslation('common');
-  const dispatch = useDispatch();
-  const [unit, setUnit] = useState<RecursiveInstitutionUnit>();
-  const [isLoadingUnit, setIsLoadingUnit] = useState(false);
-
-  useEffect(() => {
-    const fetchDepartment = async () => {
-      setIsLoadingUnit(true);
-      // TODO: NP-844 should ensure we have URIs from start (not IDs)
-      const isSubunit = orgunitId.includes('.');
-      const unitUri = isSubunit
-        ? `${CRISTIN_UNITS_BASE_URL}${orgunitId}`
-        : `${CRISTIN_INSTITUTIONS_BASE_URL}${orgunitId}`;
-      const response = await getDepartment(unitUri);
-
-      if (response?.error) {
-        dispatch(setNotification(response.error, NotificationVariant.Error));
-      } else {
-        if (!isSubunit) {
-          // Remove subunits from institution, since we only care about top-level in this case
-          delete response.subunits;
-        }
-        setUnit(response);
-      }
-      setIsLoadingUnit(false);
-    };
-
-    fetchDepartment();
-  }, [dispatch, orgunitId]);
+  const [unit, isLoadingUnitHierarchy] = useFetchUnitHierarchy(orgunitId);
 
   return (
     <StyledCard data-testid="institution-presentation">
-      {isLoadingUnit ? (
+      {isLoadingUnitHierarchy ? (
         <Progress />
       ) : (
         <>
-          <StyledTextContainer>
-            <Label>{unit?.name}</Label>
-            {unit?.subunits && <UnitRow unit={unit.subunits[0]} />}
-          </StyledTextContainer>
+          <StyledTextContainer>{unit && <AffiliationHierarchy unit={unit} />}</StyledTextContainer>
```

Comment:
burde vel kanskje brukt NormalText istedenfor divs?

Reply:
AffiliationHierarchy bruker NormalText da (trengs feks for linjekift for hvert nivå også), så det ville blitt nested NormalText-er